Marlin/stepper.cpp

```diff
@@ -796,6 +819,104 @@ void Stepper::isr() {
   #endif
 }
 
+void Stepper::chunk_steps() {
+  const uint8_t *cur_chunk = chunk_buffer[current_block->chunk_idx];
+  const uint8_t cur_chunk_block = uint8_t(step_events_completed >> 3) << 1;
+  const uint8_t block_steps = uint8_t(step_events_completed & 0x7);
+
+  const uint8_t a = cur_chunk[cur_chunk_block];
+  const uint8_t b = cur_chunk[cur_chunk_block + 1];
+
+  const uint8_t dX = a >> 4;
+  const uint8_t dY = a & 0xF;
+  const uint8_t dZ = b >> 4;
+  const uint8_t dE = b & 0xF;
+
+  uint8_t steps[4] = { 0 };
+  steps[X_AXIS] = block_moves[dX][(block_steps + 0) & 0x7];
+  steps[Y_AXIS] = block_moves[dY][(block_steps + 2) & 0x7];
+  steps[Z_AXIS] = block_moves[dZ][(block_steps + 4) & 0x7];
+  steps[E_AXIS] = block_moves[dE][(block_steps + 6) & 0x7];
+
+  //start of block, check direction
+  if(block_steps == 0) {
+    unsigned char dm = last_direction_bits;
+
+    #define UPDATE_DIR(AXIS) \
+	  if(d## AXIS == 0) {} \
```

Comment:
this line is plain wrong. Should be 

```
d## AXIS == 7
```

Just so no bit is set or removed for direction. Index 7 is 0 steps (dX = index - 7).